### ISSUE TITLE
Adapt code based on kibana version constraints to use spec version

### ISSUE THREAD

**Issue by jsoriano** [elastic/elastic-package#1342]
We are going to start relying on the spec version to determine the compatibility between packages and the stack, instead of using the kibana version as we do now.

We have to assess where the kibana version constraint is used and try to change to the new model instead.

**Comment by jlind23** [comment]
@kpollich @jsoriano isn't this something we should pursue or did I miss something?

**Comment by jsoriano** [comment]
Yes, we should do this, this is planned for the current sprint. This is not so urgent because we don't have any package relying only on the spec version for its availability yet.
